crates/uv-resolver/src/lock.rs

```diff
@@ -1813,6 +1885,76 @@ impl Dependency {
         }
     }
 
+    /// Convert the [`Dependency`] to a [`Requirement`] that can be used for resolution.
+    pub(crate) fn into_requirement(
+        self,
+        workspace_root: &Path,
+        extras: &mut HashMap<PackageName, Vec<ExtraName>>,
```

Comment:
Should we use `FxHashMap` for consistency?

Reply:
Tempted to have this return an enum (requirement or extra) rather than mutating this input argument, maybe a bit clearer. Or even just doing this at the call site? I defer to you though.

Reply:
`into_requirement` is used in three different iterator chains, so I think things are cleaner this way.